### PR TITLE
Feature/news filters active 218

### PR DESCRIPTION
**Pull request by lukewhitmore**
Added active class to filter items, based on value in URL parameter.

https://github.com/wellcometrust/upd/issues/218

### CODE REVIEW

File: www/themes/upd/upd.theme [hunk 1]
```diff
@@ -213,15 +213,29 @@ function upd_form_alter(&$form, FormStateInterface $form_state, $form_id) {
...
+
     $html = '<div class="wrapper  wrapper--gutter news-filter-container"><h4>Filter by</h4>';
+
     // Only get options which are taxonomy terms, e.g. not "all".
     foreach ($form['field_news_category_target_id']['#options'] as $k => $option) {
       $article_count = is_numeric($k) ? _upd_get_article_count($k) : NULL;
+
+      // If term is selected, set active class.
+      if ($k == $field_news_category_target_id) {
+        $state_class = ' active';
```
Comment (alfraser86) [inline www/themes/upd/upd.theme]: Maybe use a ternary for readability?
````
$state_class = $k == $field_news_category_target_id ? 'active' : '';

File: www/themes/upd/upd.theme [hunk 2]
```diff
@@ -213,15 +213,29 @@ function upd_form_alter(&$form, FormStateInterface $form_state, $form_id) {
...
+      }
+      else {
+        $state_class = '';
+      }
+
       if ($k == "All") {
-        $html .= '<a href="#" class="news-cat-changer active" data-term-id="' . $k . '">All</a>';
+        $html .= '<a href="#" class="news-cat-changer' . $state_class . '" data-term-id="' . $k . '">All</a>';
```
Comment (alfraser86) [inline www/themes/upd/upd.theme]: No L function for D8? always feel it's better than concatenating string.

**Comment by lukewhitmore** [comment]
Hey @alfraser86, I've refactored this to use D8 conventions, also reworked it so the solution works without JS (previously would only work with JS on).

File: www/themes/upd/upd.theme [hunk 3]
```diff
@@ -213,16 +215,52 @@ function upd_form_alter(&$form, FormStateInterface $form_state, $form_id) {
...
-      if ($k == "All") {
-        $html .= '<a href="#" class="news-cat-changer active" data-term-id="' . $k . '">All</a>';
-      }
-      elseif ($article_count > 0) {
-        $html .= '<a href="#" class="news-cat-changer" data-term-id="' . $k . '">' . $option . ' (' . $article_count . ')</a>';
+
+      if ($article_count > 0 || $k == 'All') {
+
+        // If term is selected, set active class.
+        if ($k == $field_news_category_target_id) {
```
Comment (alfraser86) [inline www/themes/upd/upd.theme]: ternary :-)

Comment (lukewhitmore) [reply www/themes/upd/upd.theme]: Gah, sorry .. fixed!

Comment (lukewhitmore) [reply www/themes/upd/upd.theme]: In my defence; Drupal coding standards :) .. *ducks*

**Pull request merged**